File: Selenium/Project/CRM_Activity4.py
# Logging into the site
# Goal: Open the site and login with the credentials provided

# Import webdriver from selenium
from selenium import webdriver
from selenium.webdriver.common.by import By

# Start the Driver
with webdriver.Firefox() as driver:
# Navigate to the URL
    driver.get("http://alchemy.hguy.co/crm")
    driver.maximize_window()

# Find the username field
    username = driver.find_element(By.ID, "user_name")
# Find the password field
    password = driver.find_element(By.ID, "username_password")

# Enter the given credentials
# Enter username
    username.send_keys("admin")
# Enter password
    password.send_keys("pa$$w0rd")

#Find the submit button and click it
    submit = driver.find_element(By.XPATH, "//input[@type='submit']")
    submit.click()

# Print the login message
    homePageHeader = driver.find_element(By.ID,"tab0").text
# print(homePageHeader)
if(homePageHeader=="SUITECRM DASHBOARD"):
        print("Home Page Opened Successfully !")
else:
        print("Login Failed !")

    #close the browser
driver.quit()

File: Selenium/Project/CRM_Activity4.java
package Project;

import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.testng.Assert;
import org.openqa.selenium.firefox.FirefoxDriver;
import org.testng.annotations.AfterClass;
import org.testng.annotations.BeforeClass;
import org.testng.annotations.Test;

//Enter login credentials into the respective fields
//Verify that the Homepage has opened
public class CRM_Activity4 {

	WebDriver driver;

	@BeforeClass
	public void setUp() {
		// Initialize Webdriver
		driver = new FirefoxDriver();
		// Open the page
		driver.get("http://alchemy.hguy.co/crm");
		driver.manage().window().maximize();
	}

	@Test(priority = 1)
	public void pageTitleTest() {
		// Get the page title and verify it
		Assert.assertEquals(driver.getTitle(), "SuiteCRM", "Wrong Website");

	}

	@Test(priority = 2)
	public void loginFormTest() {
		// Find the login input fields - username and password
		driver.findElement(By.id("user_name")).sendKeys("admin");
		driver.findElement(By.id("username_password")).sendKeys("pa$$w0rd");

		// Find the submit button and click it
		driver.findElement(By.xpath("//input[@type='submit']")).click();

		// verify that the homepage has opened
		String expectedUrl = "https://alchemy.hguy.co/crm/index.php?module=Home&action=index";
		String actualUrl = driver.getCurrentUrl();
		Assert.assertEquals(actualUrl, expectedUrl, "Login Failed");

	}

	@AfterClass
	public void browserClose() {
		// close the browser
		driver.quit();
	}
}
